src/utils/theme.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import _ from 'lodash/fp'
+import F from 'futil-js'
+
+export let ThemeContext = React.createContext()
+
+let hasNested = _.curry(
+  (key, theme) =>
+    key && theme && F.findIndexed((v, k) => _.startsWith(`${key}.`, k), theme)
```

Comment:
findIndex should handle empty/null things, so you shouldn't need to check `theme`, and we shouldn't even enter in to this branch if there is no `key`. Since this is curried anyway and only used here, you could simplify this to:

```js
let hasNested = key => F.findIndexed((v, k) => _.startsWith(`${key}.`, k))